Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesInterface.h

```diff
@@ -319,27 +347,91 @@ namespace AzFramework
+    using EntityPtrList = AZStd::vector<AZ::Entity*>;
+
+    struct SpawnableInstanceDescriptor
     {
-        AZ::Data::AssetId m_assetId; ///< AssetId of the spawnable
-        Spawnable::SpawnableInstanceId m_spawnableInstanceId; ///< UUid of the unique instantiated spawnable
 
-        SpawnableInstanceAddress()
-            : m_assetId()
-            , m_spawnableInstanceId(Spawnable::SpawnableInstanceId::CreateNull())
+        SpawnableInstanceDescriptor();
+
+        SpawnableInstanceDescriptor(const AZ::Data::AssetId& assetId, AZ::u32 entitySpawnTicketId); // it also generates m_spawnableInstanceId
+        
+        bool IsValid() const
         {
+            return m_assetId.IsValid() && !m_spawnableInstanceId.IsNull();
         }
 
-        SpawnableInstanceAddress(const AZ::Data::AssetId& assetId, const Spawnable::SpawnableInstanceId& spawnableInstanceId)
-            : m_assetId(assetId)
-            , m_spawnableInstanceId(spawnableInstanceId)
+        SpawnableInstanceId GetInstanceId() const
         {
+            return m_spawnableInstanceId;
         }
 
-        bool IsValid() const
+        void SetInstanceId(SpawnableInstanceId value)
         {
-            return m_assetId.IsValid() && !m_spawnableInstanceId.IsNull();
+            m_spawnableInstanceId = value;
+        }
+
+        void SetEntityIdMap(const EntityIdToEntityIdMap& baseToNewEntityIdMap)
+        {
+            m_entityIdToBaseCache.clear();
+            m_baseToNewEntityIdMap = baseToNewEntityIdMap;
         }
+
+        const EntityIdToEntityIdMap& GetEntityIdMap() const
+        {
+            return m_baseToNewEntityIdMap;
+        }
+
+        const EntityIdToEntityIdMap& GetEntityIdToBaseMap() const //reverse map
+        {
+            if (m_entityIdToBaseCache.empty())
+            {
+                BuildReverseLookUp();
+            }
+            return m_entityIdToBaseCache;
+        }
+
+        const AZ::Data::AssetId& GetAssetId() const
+        {
+            return m_assetId;
+        }
+
+        const EntityPtrList& GetInstantiatedEntities() const
+        {
+            return m_entityPtrList;
+        }
+
+        void SetInstantiatedEntities(const EntityPtrList& entityList)
+        {
+            m_entityPtrList = entityList;
+        }
+
+        const AZ::u32 GetEntitySpawnTicketId() const
+        {
+            return m_entitySpawnTicketId;
+        }
+
+        static AZStd::shared_ptr<SpawnableInstanceDescriptor> GetInvalidDescriptor();
+
+        bool FinalizeCreateInstance(void* remapContainer,
+            const AZ::Uuid& classUuid,
+            const AZ::IdUtils::Remapper<AZ::EntityId>::IdMapper& customMapper);
+
+    protected:
+
+        AZ::Data::AssetId m_assetId; ///< AssetId of the spawnable
+        SpawnableInstanceId m_spawnableInstanceId; ///< UUid of the unique instantiated spawnable
+        EntityIdToEntityIdMap m_baseToNewEntityIdMap; ///< Map of old entityId to new
```

Comment:
Let the comment be "Map of the static entityId to the new (dynamic) entityId"

Reply:
Updated